In [1]:
# Import Pandas library for data wrangling / data munging
import pandas as pd

In [2]:
# Read the csv file using pd.read_csv() method
employee = pd.read_csv('employee_data.csv')
insurance = pd.read_csv('insurance_data.csv')
vendor = pd.read_csv('vendor_data.csv')

In [3]:
# head() method gives top 5 rows of our dataframe
employee.head()

AGENT_ID      AGENT_NAME DATE_OF_JOINING          ADDRESS_LINE1  \
0  AGENT00001       Ray Johns      1993-06-05       1402 Maggies Way   
1  AGENT00002   Angelo Borjon      2005-12-27         414 Tanya Pass   
2  AGENT00003  Candy Spellman      2003-09-02    606 National Street   
3  AGENT00004      Mary Smith      2004-09-23  235 Hugh Thomas Drive   
4  AGENT00005    Mildred Diaz      2011-06-21  3426 Broadview Street   

  ADDRESS_LINE2              CITY STATE  POSTAL_CODE  EMP_ROUTING_NUMBER  \
0           NaN  Waterbury Center    VT         5677            34584958   
1           NaN       Panama City    FL        32404           107363763   
2          #306      Fayetteville    AR        72701            81744097   
3           NaN       Panama City    FL        32404            67563771   
4           NaN        Montgomery    AL        36110           114951317   

      EMP_ACCT_NUMBER  
0  HKUN51252328472585  
1  OPIS19290040088204  
2  YSCJ67489688482590  
3  ZANG21285355574581  
4  DZFS82244494451134

In [4]:
# head() method gives top 5 rows of our dataframe
insurance.head()

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
0  2020-06-01 00:00:00    TXN00000001   A00003822   PLC00008468    2015-06-23   
1  2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594    2018-04-21   
2  2020-06-01 00:00:00    TXN00000003   A00003172   PLC00007969    2019-10-03   
3  2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292    2016-11-29   
4  2020-06-01 00:00:00    TXN00000005   A00008173   PLC00000204    2011-12-26   

      LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  ...  \
0  2020-05-16  2020-05-21         Health          157.13          9000  ...   
1  2020-05-13  2020-05-18       Property          141.71         26000  ...   
2  2020-05-21  2020-05-26       Property          157.24         13000  ...   
3  2020-05-14  2020-05-19         Health          172.87         16000  ...   
4  2020-05-17  2020-05-22         Travel           88.53          3000  ...   

  CLAIM_STATUS INCIDENT_SEVERITY AUTHORITY_CONTACTED ANY_INJURY  \
0            A        Major Loss              Police          0   
1            A        Total Loss           Ambulance          1   
2            A        Total Loss              Police          0   
3            A        Minor Loss           Ambulance          0   
4            A        Major Loss              Police          0   

  POLICE_REPORT_AVAILABLE  INCIDENT_STATE   INCIDENT_CITY  \
0                       1              GA        Savannah   
1                       0              AL      Montgomery   
2                       1              CO  Grand Junction   
3                       0              GA        Savannah   
4                       1              TN       Nashville   

  INCIDENT_HOUR_OF_THE_DAY    AGENT_ID  VENDOR_ID  
0                        4  AGENT00413  VNDR00556  
1                        0  AGENT00769  VNDR00592  
2                       19  AGENT00883  VNDR00031  
3                       12  AGENT00278  VNDR00075  
4                       18  AGENT00636  VNDR00472  

[5 rows x 38 columns]

In [5]:
# head() method gives top 5 rows of our dataframe
vendor.head()

VENDOR_ID              VENDOR_NAME                   ADDRESS_LINE1  \
0  VNDR00001  King, Proctor and Jones        2027 North Shannon Drive   
1  VNDR00002               Garcia Ltd          5701 East Shirley Lane   
2  VNDR00003               Cherry LLC            1217 Cottondale Road   
3  VNDR00004              Mays-Benson  227 West Montgomery Cross Road   
4  VNDR00005               Wilson PLC            23 North Hill Street   

  ADDRESS_LINE2          CITY STATE  POSTAL_CODE  
0            #5  Fayetteville    AR        72703  
1           NaN    Montgomery    AL        36117  
2           NaN    Montgomery    AL        36109  
3          #736      Savannah    GA        31406  
4           NaN     Nashville    TN        37210

# Task # 1
* Merge the 3 dataset and create 1 view of data.
* You can merge insurance_data.csv and employee_data.csv on
* AGENT_ID You can merge insurance_data.csv and vendor_data.csv on
* VENDOR_ID Note: Use left Outer join as not all claims require Vendor

In [6]:
# Using pd.merge() method we can merge all datasets in one dataset using a left outer join
merged=pd.merge(insurance,employee,on='AGENT_ID',how='left',suffixes=('_CUSTOMER','_AGENT'))
view=pd.merge(merged,vendor,on='VENDOR_ID',how='left')
view.head()

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
0  2020-06-01 00:00:00    TXN00000001   A00003822   PLC00008468    2015-06-23   
1  2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594    2018-04-21   
2  2020-06-01 00:00:00    TXN00000003   A00003172   PLC00007969    2019-10-03   
3  2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292    2016-11-29   
4  2020-06-01 00:00:00    TXN00000005   A00008173   PLC00000204    2011-12-26   

      LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  ...  \
0  2020-05-16  2020-05-21         Health          157.13          9000  ...   
1  2020-05-13  2020-05-18       Property          141.71         26000  ...   
2  2020-05-21  2020-05-26       Property          157.24         13000  ...   
3  2020-05-14  2020-05-19         Health          172.87         16000  ...   
4  2020-05-17  2020-05-22         Travel           88.53          3000  ...   

  STATE_AGENT POSTAL_CODE_AGENT EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  \
0          TN             37217           35441155  LRZO53254369804934   
1          AZ             85301           44055451  XBYG14382904542874   
2          AL             36105           55770246  WRXH14597428306333   
3          OK             73071           87652976  YEPN16077902472499   
4          MD             21061           76119019  HZBK86544578578547   

                        VENDOR_NAME              ADDRESS_LINE1 ADDRESS_LINE2  \
0                        Ramsey Inc       3114 West 20th Court           NaN   
1  Morales, Campbell and Washington      42 West Louise Street             B   
2                       Johnson Ltd    2619 North Quality Lane          #315   
3                  Blackwell-Foster          3609 Meadow Drive           NaN   
4                       Evans-Scott  4001 19th Place Northeast           NaN   

           CITY  STATE  POSTAL_CODE  
0   Panama City     FL      32405.0  
1  Fayetteville     AR      72701.0  
2  Fayetteville     AR      72703.0  
3     Nashville     TN      37215.0  
4    Washington     DC      20018.0  

[5 rows x 53 columns]

# Task #2
Business Leader wants to find Top 3 Insurance Type where we are getting
most insurance claims

In [7]:
# Fetch values of top 3 insurance types for most insurance claims
insurance['INSURANCE_TYPE'].value_counts()[:3]

Property    1692
Mobile      1692
Health      1690
Name: INSURANCE_TYPE, dtype: int64

Conclusion Drawn:-

* From the above dataframe we get to know that there is a equal proportion between Property as well Mobile for insurance claims whereas Health type has least claims as compared to both of them.

# Task #3
* Business Leader wants to find Top 5 States where we are getting most insurance claims
for customer belonging to HIGH(H) risk segment?

In [10]:
# Fetch top 5 states with most insurance claims according to their high risk segmentation
insurance[insurance['RISK_SEGMENTATION']=='H'].STATE.value_counts()[:5]

CA    148
AZ    105
FL    104
TN    100
AR     97
Name: STATE, dtype: int64

Conclusion Drawn:-

* State CA has the highest most insurance claims with higher risk and State AR has the least insurance claims with low risk.

# Task #4
* Business wants to create a new variable “COLOCATION” which will have following values IF
* Customer State == Incident State == Agent Address State THEN 1 ELSE 0
* Find the mean of this new column

In [11]:
# Fetch the features of view
view.columns

Index(['TXN_DATE_TIME', 'TRANSACTION_ID', 'CUSTOMER_ID', 'POLICY_NUMBER',
       'POLICY_EFF_DT', 'LOSS_DT', 'REPORT_DT', 'INSURANCE_TYPE',
       'PREMIUM_AMOUNT', 'CLAIM_AMOUNT', 'CUSTOMER_NAME',
       'ADDRESS_LINE1_CUSTOMER', 'ADDRESS_LINE2_CUSTOMER', 'CITY_CUSTOMER',
       'STATE_CUSTOMER', 'POSTAL_CODE_CUSTOMER', 'SSN', 'MARITAL_STATUS',
       'AGE', 'TENURE', 'EMPLOYMENT_STATUS', 'NO_OF_FAMILY_MEMBERS',
       'RISK_SEGMENTATION', 'HOUSE_TYPE', 'SOCIAL_CLASS', 'ROUTING_NUMBER',
       'ACCT_NUMBER', 'CUSTOMER_EDUCATION_LEVEL', 'CLAIM_STATUS',
       'INCIDENT_SEVERITY', 'AUTHORITY_CONTACTED', 'ANY_INJURY',
       'POLICE_REPORT_AVAILABLE', 'INCIDENT_STATE', 'INCIDENT_CITY',
       'INCIDENT_HOUR_OF_THE_DAY', 'AGENT_ID', 'VENDOR_ID', 'AGENT_NAME',
       'DATE_OF_JOINING', 'ADDRESS_LINE1_AGENT', 'ADDRESS_LINE2_AGENT',
       'CITY_AGENT', 'STATE_AGENT', 'POSTAL_CODE_AGENT', 'EMP_ROUTING_NUMBER',
       'EMP_ACCT_NUMBER', 'VENDOR_NAME', 'ADDRESS_LINE1', 'ADDRESS_LINE2',
       

In [12]:
# Create new feature 'COLOCATION' with values such as customer state, incident state, agent address state
view['COLOCATION'] = (view['STATE_CUSTOMER']==view['INCIDENT_STATE'])&(view['INCIDENT_STATE']==view['STATE_AGENT'])
view['COLOCATION']

0       False
1       False
2       False
3       False
4       False
        ...  
9995    False
9996    False
9997    False
9998    False
9999    False
Name: COLOCATION, Length: 10000, dtype: bool

In [13]:
# Converting boolean values of 'COLOCATION' feature into integer values
view['COLOCATION']=view.COLOCATION.astype('int')
view['COLOCATION']

0       0
1       0
2       0
3       0
4       0
       ..
9995    0
9996    0
9997    0
9998    0
9999    0
Name: COLOCATION, Length: 10000, dtype: int32

In [14]:
# Fetch the 'COLOCATION' mean value
view['COLOCATION'].mean()

0.0044

* Conclusion Drawn:-

* The mean for the feature 'COLOCATION' is 0.0044.

# Task #5
Data entry error was detected in the data and you are required to correct it. If for any
claim transaction “AUTHORITY_CONTACTED” is NOT “Police” and POLICE_AVAILABLE == 1
Then Update “AUTHORITY_CONTACTED” to Police.


In [15]:
# Create feature with AUTHORITY_CONTACTED not equal to Police
# Fetch features for the feature 'not_police'
not_police = view[view['AUTHORITY_CONTACTED']!='Police']
not_police.columns

Index(['TXN_DATE_TIME', 'TRANSACTION_ID', 'CUSTOMER_ID', 'POLICY_NUMBER',
       'POLICY_EFF_DT', 'LOSS_DT', 'REPORT_DT', 'INSURANCE_TYPE',
       'PREMIUM_AMOUNT', 'CLAIM_AMOUNT', 'CUSTOMER_NAME',
       'ADDRESS_LINE1_CUSTOMER', 'ADDRESS_LINE2_CUSTOMER', 'CITY_CUSTOMER',
       'STATE_CUSTOMER', 'POSTAL_CODE_CUSTOMER', 'SSN', 'MARITAL_STATUS',
       'AGE', 'TENURE', 'EMPLOYMENT_STATUS', 'NO_OF_FAMILY_MEMBERS',
       'RISK_SEGMENTATION', 'HOUSE_TYPE', 'SOCIAL_CLASS', 'ROUTING_NUMBER',
       'ACCT_NUMBER', 'CUSTOMER_EDUCATION_LEVEL', 'CLAIM_STATUS',
       'INCIDENT_SEVERITY', 'AUTHORITY_CONTACTED', 'ANY_INJURY',
       'POLICE_REPORT_AVAILABLE', 'INCIDENT_STATE', 'INCIDENT_CITY',
       'INCIDENT_HOUR_OF_THE_DAY', 'AGENT_ID', 'VENDOR_ID', 'AGENT_NAME',
       'DATE_OF_JOINING', 'ADDRESS_LINE1_AGENT', 'ADDRESS_LINE2_AGENT',
       'CITY_AGENT', 'STATE_AGENT', 'POSTAL_CODE_AGENT', 'EMP_ROUTING_NUMBER',
       'EMP_ACCT_NUMBER', 'VENDOR_NAME', 'ADDRESS_LINE1', 'ADDRESS_LINE2',
       

In [16]:
# Update observations with POLICE_REPORT_AVAILABLE with their Index values
updated_records= not_police[not_police['POLICE_REPORT_AVAILABLE']==1].index
updated_records.shape

(4870,)

In [17]:
# Update the Authority feature
view.loc[updated_records,'AUTHORITY_CONTACTED'] = 'Police'

In [18]:
# Check errors in the data now after correcting it
view[(view['POLICE_REPORT_AVAILABLE']==1)&(view['AUTHORITY_CONTACTED']!='Police')]

Empty DataFrame
Columns: [TXN_DATE_TIME, TRANSACTION_ID, CUSTOMER_ID, POLICY_NUMBER, POLICY_EFF_DT, LOSS_DT, REPORT_DT, INSURANCE_TYPE, PREMIUM_AMOUNT, CLAIM_AMOUNT, CUSTOMER_NAME, ADDRESS_LINE1_CUSTOMER, ADDRESS_LINE2_CUSTOMER, CITY_CUSTOMER, STATE_CUSTOMER, POSTAL_CODE_CUSTOMER, SSN, MARITAL_STATUS, AGE, TENURE, EMPLOYMENT_STATUS, NO_OF_FAMILY_MEMBERS, RISK_SEGMENTATION, HOUSE_TYPE, SOCIAL_CLASS, ROUTING_NUMBER, ACCT_NUMBER, CUSTOMER_EDUCATION_LEVEL, CLAIM_STATUS, INCIDENT_SEVERITY, AUTHORITY_CONTACTED, ANY_INJURY, POLICE_REPORT_AVAILABLE, INCIDENT_STATE, INCIDENT_CITY, INCIDENT_HOUR_OF_THE_DAY, AGENT_ID, VENDOR_ID, AGENT_NAME, DATE_OF_JOINING, ADDRESS_LINE1_AGENT, ADDRESS_LINE2_AGENT, CITY_AGENT, STATE_AGENT, POSTAL_CODE_AGENT, EMP_ROUTING_NUMBER, EMP_ACCT_NUMBER, VENDOR_NAME, ADDRESS_LINE1, ADDRESS_LINE2, CITY, STATE, POSTAL_CODE, COLOCATION]
Index: []

[0 rows x 54 columns]

Conclusion Drawn:-

From the above dataframe we can conclude there are no more errors in our dataset.

# Task #6
* Business wants to check the Claim Amount for deviation for each transaction, they would
like you to calculate as follow
* CLAIM_DEVIATION = AVG_CLAIM_AMOUNT_FOR_LAST_30DAYS (same insurance type)
/ CURRENT_CLAIM_AMOUNT
* If the value < 0.5 THEN CLAIM_DEVIATION = 1 ELSE 0
* If there is less than 30 days of transaction history THEN
-1 Note: LAST_30DAYS does not include current day

In [19]:
# Fetch max value for TXN_DATE_TIME feature
view.TXN_DATE_TIME.max()

'2021-06-30 00:00:00'

In [20]:
# Converting TXN_DATE_TIME feature max value into a datetime form
todays_date = pd.to_datetime('2021-06-30')
todays_date

Timestamp('2021-06-30 00:00:00')

In [21]:
# Look through the values of feature 'TXN_DATE-TIME' and converting it into a datetime form
view['TXN_DATE_TIME'] = pd.to_datetime(view['TXN_DATE_TIME'])
view['TXN_DATE_TIME']

0      2020-06-01
1      2020-06-01
2      2020-06-01
3      2020-06-01
4      2020-06-01
          ...    
9995   2021-06-30
9996   2021-06-30
9997   2021-06-30
9998   2021-06-30
9999   2021-06-30
Name: TXN_DATE_TIME, Length: 10000, dtype: datetime64[ns]

In [22]:
# Fetch last 30 days value from feature 'TXN_DATE_TIME'
last_30 = view.loc[(view['TXN_DATE_TIME'] > '2021-05-30') & (view['TXN_DATE_TIME'] < '2021-06-30')] 
last_30

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
9189    2021-05-31    TXN00009190   A00004648   PLC00005883    2014-12-20   
9190    2021-05-31    TXN00009191   A00005798   PLC00001906    2015-02-02   
9191    2021-05-31    TXN00009192   A00009768   PLC00000089    2017-12-21   
9192    2021-05-31    TXN00009193   A00005540   PLC00001060    2015-01-18   
9193    2021-05-31    TXN00009194   A00001966   PLC00006779    2012-04-10   
...            ...            ...         ...           ...           ...   
9970    2021-06-29    TXN00009971   A00005227   PLC00004359    2010-09-20   
9971    2021-06-29    TXN00009972   A00002887   PLC00003785    2020-01-16   
9972    2021-06-29    TXN00009973   A00009361   PLC00001736    2014-12-30   
9973    2021-06-29    TXN00009974   A00002159   PLC00006446    2016-08-12   
9974    2021-06-29    TXN00009975   A00006567   PLC00002837    2015-08-27   

         LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  \
9189  2021-05-28  2021-05-28       Property          112.57         16000   
9190  2021-05-24  2021-05-26         Travel           69.41          2000   
9191  2021-05-23  2021-05-26          Motor           87.40          5000   
9192  2021-05-23  2021-05-26       Property          131.05         12000   
9193  2021-05-11  2021-05-16       Property           92.45         31000   
...          ...         ...            ...             ...           ...   
9970  2021-06-27  2021-06-27         Travel           54.14          5000   
9971  2021-06-17  2021-06-22           Life           67.11         81000   
9972  2021-06-14  2021-06-19         Travel           63.49          5000   
9973  2021-06-27  2021-06-27         Mobile            9.22           600   
9974  2021-06-23  2021-06-25          Motor          115.60         10000   

      ... POSTAL_CODE_AGENT EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  \
9189  ...             95501          114787167  NPRV40932107822625   
9190  ...              6042          109301503  EMFD60069652268552   
9191  ...              2721           36004986  IXEZ51619621113989   
9192  ...              1863           77423296  TEUS71549591731930   
9193  ...             72703           55071686  WURP39860255825021   
...   ...               ...                ...                 ...   
9970  ...             37027           52980550  PBOE57461807615642   
9971  ...             20373           28060187  TDQU90289065114883   
9972  ...              6040          124190333  YFFY90307447508970   
9973  ...              6040           71906146  WING80506589957019   
9974  ...             20778           70250107  XRKS19473964978781   

                     VENDOR_NAME                  ADDRESS_LINE1  \
9189                  Wilson PLC           23 North Hill Street   
9190     Welch, Keller and Hicks     1820 22nd Street Southeast   
9191                         NaN                            NaN   
9192                    Rose Ltd           1123 Glenwood Avenue   
9193                         NaN                            NaN   
...                          ...                            ...   
9970            Bartlett-Winters                  113 Dove Lane   
9971   Mendoza, Moran and Parker              6463 Vrain Street   
9972     Reeves, Brown and Burch  467 North Mountain Ranch Road   
9973                         NaN                            NaN   
9974  Meadows, Wiggins and Lewis             50 Winthrop Avenue   

      ADDRESS_LINE2          CITY STATE  POSTAL_CODE  COLOCATION  
9189            NaN     Nashville    TN      37210.0           0  
9190            NaN    Washington    DC      20020.0           0  
9191            NaN           NaN   NaN          NaN           0  
9192            NaN     Nashville    TN      37204.0           0  
9193            NaN           NaN   NaN          NaN           0  
...             ...           ...   ...          ...         ...  
9970            NaN      Savannah    GA 

In [23]:
# Fetch average claim amount from last 30 days by grouping with INSURANCE_TYPE and converting it into a dictionary
avg_claim = last_30.groupby('INSURANCE_TYPE').mean()['CLAIM_AMOUNT'].to_dict()
avg_claim

{'Health': 10904.411764705883,
 'Life': 52242.857142857145,
 'Mobile': 378.35820895522386,
 'Motor': 5573.770491803279,
 'Property': 24627.906976744187,
 'Travel': 2800.0}

Conclusion Drawn:-

From the above dataframe we get to know that Life insurance type has the highest average claim amount whereas Mobile insurance type has least average claim amount.

In [25]:
# define a function for calculating claim_deviation
def Claim_deviation(row):
    if (todays_date - row['TXN_DATE_TIME'])<pd.Timedelta(days=30):
        return -1
    claim_deviation = avg_claim[row['INSURANCE_TYPE']] / row['CLAIM_AMOUNT']
    if claim_deviation < 0.5:
        return 1
    else:
        return 0

In [26]:
# Applying the function to the dataframe
view['CLAIM_DEVIATION'] = view.apply(Claim_deviation,axis=1)
view['CLAIM_DEVIATION']

0       0
1       0
2       0
3       0
4       0
       ..
9995   -1
9996   -1
9997   -1
9998   -1
9999   -1
Name: CLAIM_DEVIATION, Length: 10000, dtype: int64

In [27]:
# Fetch values for feature CLAIM_DEVIATION
view['CLAIM_DEVIATION'].value_counts()

 0    9216
-1     784
Name: CLAIM_DEVIATION, dtype: int64

Conclusion Drawn:-

For values less than 0.5 there are 9216 claims whereas 784 claims with transaction history of less than 30 days.

# Task #7
Find All Agents who have worked on more than 2 types of Insurance Claims. Sort them by
Total Claim Amount Approved under them in descending order

In [30]:
# Calculating insurance types worked on of each agent
type_level_1 = view.groupby('AGENT_ID')['INSURANCE_TYPE'].nunique()
type_level_1

AGENT_ID
AGENT00001    4
AGENT00002    2
AGENT00003    6
AGENT00004    6
AGENT00005    4
             ..
AGENT01196    4
AGENT01197    4
AGENT01198    4
AGENT01199    5
AGENT01200    6
Name: INSURANCE_TYPE, Length: 1200, dtype: int64

In [31]:
# Calculating whether agent has more than 2 insurance types
agent_insurance_types = type_level_1>2
agent_insurance_types.value_counts()

True     1164
False      36
Name: INSURANCE_TYPE, dtype: int64

In [32]:
# Getting list of agents who have worked on more than 2 types of insurance types
multi_level_agents = agent_insurance_types[agent_insurance_types].index
multi_level_agents.shape

(1164,)

In [33]:
# Fetch top level agents by grouping AGENT_ID, AGENT_NAME with their claim amounts and sorting them into a descending order
Top_level_agents = view.groupby(['AGENT_ID','AGENT_NAME']).sum().loc[multi_level_agents][['CLAIM_AMOUNT']].sort_values('CLAIM_AMOUNT',ascending=False)
Top_level_agents

CLAIM_AMOUNT
AGENT_ID   AGENT_NAME                    
AGENT00807 Don Filkins             528800
AGENT00679 Clara Barnett           489000
AGENT00771 Roger Burns             422100
AGENT00125 Salvador Soriano        400400
AGENT00789 Alison Baron            392900
...                                   ...
AGENT00910 Vincent Ragan            15500
AGENT00689 Connie Hunt              13700
AGENT00732 Sylvia Tran              11300
AGENT00706 Laura Staggs             10800
AGENT00604 Thelma Salinas            5500

[1164 rows x 1 columns]

In [34]:
# Sorting agents who have worked on more than 2 types of Insurance Claims with their Total Claim Amount in descending order
Top_level_agents.join(type_level_1)

CLAIM_AMOUNT  INSURANCE_TYPE
AGENT_ID   AGENT_NAME                                    
AGENT00807 Don Filkins             528800               6
AGENT00679 Clara Barnett           489000               4
AGENT00771 Roger Burns             422100               5
AGENT00125 Salvador Soriano        400400               4
AGENT00789 Alison Baron            392900               4
...                                   ...             ...
AGENT00910 Vincent Ragan            15500               3
AGENT00689 Connie Hunt              13700               3
AGENT00732 Sylvia Tran              11300               3
AGENT00706 Laura Staggs             10800               3
AGENT00604 Thelma Salinas            5500               3

[1164 rows x 2 columns]

# Task #8
* Mobile & Travel Insurance premium are discounted by 10%
* Health and Property Insurance premium are increased by 7%
* Life and Motor Insurance premium are marginally increased by 2%
* What will be overall change in % of the Premium Amount Collected for all these Customer?

In [36]:
# Fetch the premium value collected
premium_collected = view['PREMIUM_AMOUNT'].sum()
premium_collected

885085.95

In [37]:
# Fetch the value of premium amount based on their types and converting it into a dictionary datatype
premium_type = view.groupby('INSURANCE_TYPE').sum()['PREMIUM_AMOUNT'].to_dict()
premium_type

{'Health': 252455.73,
 'Life': 125621.22,
 'Mobile': 15191.42,
 'Motor': 165391.98,
 'Property': 202285.83,
 'Travel': 124139.77}

In [38]:
# Mobile & Travel premium discounted by 10%
premium_type['Mobile']*=0.9
premium_type['Travel']*=0.9
# Health and Property premium increases by 7%
premium_type['Health']*=1.07
premium_type['Property']*=1.07
# Life and Motor premium increased by 2%
premium_type['Life']*=1.02
premium_type['Motor']*=1.02

In [39]:
# Premium after discount and increase
new_premium = sum(premium_type.values())
new_premium

908805.0042000001

In [40]:
# Calculate average premium collected 
(new_premium-premium_collected)*100/premium_collected

2.6798588543858517

In [41]:
# Fetch the value of premium collected and value of new premium collected
premium_collected,new_premium

(885085.95, 908805.0042000001)

Conclusion Drawn:-

Overall change in % of the Premium Amount Collected for all these Customer is 2.67%.

# Task #9
* Business wants to give discount to customer who are loyal and under stress due to Covid19. They have laid down an eligibility Criteria as follow
- IF CUSTOMER_TENURE > 60 AND EMPLOYMENT_STATUS = “N”
- AND NO_OF_FAMILY_MEMBERS >=4 THEN 1 ELSE 0
- Create a new column “ELIGIBLE_FOR_DISCOUNT” and find it mean.

In [47]:
# Fetch discount status value with given conditions
discount_status=(view['TENURE']>60)&(view['EMPLOYMENT_STATUS']=='N')&(view['NO_OF_FAMILY_MEMBERS']>=4)
discount_status

0       False
1       False
2       False
3       False
4       False
        ...  
9995    False
9996    False
9997    False
9998    False
9999    False
Length: 10000, dtype: bool

In [48]:
# Create new feature and converting it into integer dtype
view['ELIGIBLE_FOR_DISCOUNT'] = discount_status.astype('int')
view['ELIGIBLE_FOR_DISCOUNT']

0       0
1       0
2       0
3       0
4       0
       ..
9995    0
9996    0
9997    0
9998    0
9999    0
Name: ELIGIBLE_FOR_DISCOUNT, Length: 10000, dtype: int32

In [49]:
# Mean of feature 'ELIGIBLE_FOR_DISCOUNT'
view['ELIGIBLE_FOR_DISCOUNT'].mean()

0.0299

Conclusion Drawn:-

The mean for the feature 'ELIGIBLE_FOR_DISCOUNT' is 0.0299.

# Task #10
* Business wants to check Claim Velocity which is defined as follow
* CLAIM_VELOCITY = NO_OF_CLAIMS_IN_LAST30DAYS (for the current insurance type)
/ NO_OF_CLAIMS_IN_LAST3DAYS (for the current insurance type)
* Note: LAST30DAYS & LAST3DAYS does not include current
day


In [51]:
# Fetch current date value
todays_date

Timestamp('2021-06-30 00:00:00')

In [52]:
# Fetch number of values in feature 'CLAIM_STATUS'
view['CLAIM_STATUS'].value_counts()

A    9497
D     503
Name: CLAIM_STATUS, dtype: int64

### We had defined last 30 days data in task # 6 earlier, similarly defining for last 3 days.

In [53]:
# Fetch values for last 3 days and converting it into timedelta
# Timedelta represents a duration, the difference between two dates or times.
last_3_days = view.loc[(view['TXN_DATE_TIME'] > todays_date-pd.Timedelta(days=4)) & (view['TXN_DATE_TIME'] < todays_date)]
last_3_days

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
9897    2021-06-27    TXN00009898   A00005727   PLC00007733    2019-11-16   
9898    2021-06-27    TXN00009899   A00003496   PLC00001925    2018-01-30   
9899    2021-06-27    TXN00009900   A00006123   PLC00008473    2010-07-25   
9900    2021-06-27    TXN00009901   A00000856   PLC00002051    2010-10-18   
9901    2021-06-27    TXN00009902   A00003636   PLC00008806    2013-09-26   
...            ...            ...         ...           ...           ...   
9970    2021-06-29    TXN00009971   A00005227   PLC00004359    2010-09-20   
9971    2021-06-29    TXN00009972   A00002887   PLC00003785    2020-01-16   
9972    2021-06-29    TXN00009973   A00009361   PLC00001736    2014-12-30   
9973    2021-06-29    TXN00009974   A00002159   PLC00006446    2016-08-12   
9974    2021-06-29    TXN00009975   A00006567   PLC00002837    2015-08-27   

         LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  \
9897  2021-06-14  2021-06-19       Property          114.82         17000   
9898  2021-06-15  2021-06-20          Motor           76.35          7000   
9899  2021-06-21  2021-06-23         Mobile            9.10           300   
9900  2021-06-25  2021-06-25           Life           67.13         68000   
9901  2021-06-08  2021-06-13         Mobile           10.95           600   
...          ...         ...            ...             ...           ...   
9970  2021-06-27  2021-06-27         Travel           54.14          5000   
9971  2021-06-17  2021-06-22           Life           67.11         81000   
9972  2021-06-14  2021-06-19         Travel           63.49          5000   
9973  2021-06-27  2021-06-27         Mobile            9.22           600   
9974  2021-06-23  2021-06-25          Motor          115.60         10000   

      ...     EMP_ACCT_NUMBER                    VENDOR_NAME  \
9897  ...  HUII66210152910986                            NaN   
9898  ...  HCQB64970746728804                   Walker-Hines   
9899  ...  QIJJ36053901151211                            NaN   
9900  ...  MNQC41165368811882                    Roberts Inc   
9901  ...  PHPT29879348970185  Sanchez, Nichols and Robinson   
...   ...                 ...                            ...   
9970  ...  PBOE57461807615642               Bartlett-Winters   
9971  ...  TDQU90289065114883      Mendoza, Moran and Parker   
9972  ...  YFFY90307447508970        Reeves, Brown and Burch   
9973  ...  WING80506589957019                            NaN   
9974  ...  XRKS19473964978781     Meadows, Wiggins and Lewis   

                      ADDRESS_LINE1 ADDRESS_LINE2          CITY  STATE  \
9897                            NaN           NaN           NaN    NaN   
9898                824 Main Street             D    Manchester     CT   
9899                            NaN           NaN           NaN    NaN   
9900         10202 North 51st Drive           NaN      Glendale     AZ   
9901      949 South Lawrence Street           NaN    Montgomery     AL   
...                             ...           ...           ...    ...   
9970                  113 Dove Lane           NaN      Savannah     GA   
9971              6463 Vrain Street           NaN        Arvada     CO   
9972  467 North Mountain Ranch Road           NaN  Fayetteville     AR   
9973                            NaN           NaN           NaN    NaN   
9974             50 Winthrop Avenue           NaN      Lawrence     MA   

     POSTAL_CODE COLOCATION  CLAIM_DEVIATION  ELIGIBLE_FOR_DISCOUNT  
9897         NaN          0               -1                      0  
9898      6040.0          0               -1                      0  
9899         NaN          0               -1                      0  
9900     85302.0          0               -1                      0  
9901     36104.0          0               -1                      0  
...          ...        ...              ...                    ...  
9970   

In [54]:
# Calculating total approved claims insurance type wise in last 30 days
claims_in_last_30_days = pd.DataFrame.from_dict(last_30[last_30['CLAIM_STATUS']=='A'].groupby('INSURANCE_TYPE').size().to_dict(),orient='index',columns=['NO_OF_CLAIMS_IN_LAST30DAYS'])
claims_in_last_30_days

NO_OF_CLAIMS_IN_LAST30DAYS
Health                           128
Life                             128
Mobile                           127
Motor                            117
Property                         125
Travel                           123

In [55]:
# Calculating total approved claims insurance type wise in last 3 days
claims_in_last_3_days = pd.DataFrame.from_dict(last_3_days[last_3_days['CLAIM_STATUS']=='A'].groupby('INSURANCE_TYPE').size().to_dict(),orient='index',columns=['NO_OF_CLAIMS_IN_LAST3DAYS'])
claims_in_last_3_days

NO_OF_CLAIMS_IN_LAST3DAYS
Health                           14
Life                             12
Mobile                            9
Motor                            12
Property                         12
Travel                           14

In [56]:
# Fetch the value of approved claims by joining claims in last 30 and 3 days
approved_claims = claims_in_last_30_days.join(claims_in_last_3_days)
approved_claims

NO_OF_CLAIMS_IN_LAST30DAYS  NO_OF_CLAIMS_IN_LAST3DAYS
Health                           128                         14
Life                             128                         12
Mobile                           127                          9
Motor                            117                         12
Property                         125                         12
Travel                           123                         14

In [57]:
# Create new feature CLAIM_VELOCITY by calculating difference between no. of claims in last 30 days and 3 days
approved_claims['CLAIM_VELOCITY'] = approved_claims['NO_OF_CLAIMS_IN_LAST30DAYS']/approved_claims['NO_OF_CLAIMS_IN_LAST3DAYS']
approved_claims['CLAIM_VELOCITY']

Health       9.142857
Life        10.666667
Mobile      14.111111
Motor        9.750000
Property    10.416667
Travel       8.785714
Name: CLAIM_VELOCITY, dtype: float64

Conclusion Drawn:-

From the above table we get to know that the CLAIM_VELOCITY for Mobile is very high and least CLAIM_VELOCITY of Travel.

# Task #11
Find all low performing agents i.e. employees who are in the bottom 5 percentile based
on Claims worked by them.

In [58]:
# Joining feature 'AGENT_ID' with 'CLAIM_AMOUNT'
bottom_5_percentile = view.groupby('AGENT_ID').sum().quantile(0.05)['CLAIM_AMOUNT']
bottom_5_percentile

30395.0

In [59]:
# Joining feature 'AGENT_ID' without mentioning quantile value now
Agent_grouped_data = view.groupby('AGENT_ID').sum()
Agent_grouped_data

PREMIUM_AMOUNT  CLAIM_AMOUNT  POSTAL_CODE_CUSTOMER  AGE  TENURE  \
AGENT_ID                                                                      
AGENT00001          678.43         85500                608359  452     716   
AGENT00002          187.48         84000                 98606  126     170   
AGENT00003          879.39        170300                550288  448     825   
AGENT00004          959.01        297600                760766  623     995   
AGENT00005          524.82         45000                 72585  221     524   
...                    ...           ...                   ...  ...     ...   
AGENT01196         1107.19        274000                520368  474     442   
AGENT01197          536.82        156600                278167  282     507   
AGENT01198          724.28        196400                304691  238     443   
AGENT01199          903.30        164700                400930  390     552   
AGENT01200          844.98        100900                345362  469     732   

            NO_OF_FAMILY_MEMBERS  ROUTING_NUMBER  ANY_INJURY  \
AGENT_ID                                                       
AGENT00001                    51       666535715           7   
AGENT00002                     9       216003092           2   
AGENT00003                    50       869300559           8   
AGENT00004                    55       918911799          11   
AGENT00005                    22       304269316           5   
...                          ...             ...         ...   
AGENT01196                    39       697583096           7   
AGENT01197                    26       461840846           3   
AGENT01198                    32       573544520           5   
AGENT01199                    37       595102223           5   
AGENT01200                    57       650329977          10   

            POLICE_REPORT_AVAILABLE  INCIDENT_HOUR_OF_THE_DAY  \
AGENT_ID                                                        
AGENT00001                       10                       104   
AGENT00002                        3                        37   
AGENT00003                        7                       112   
AGENT00004                        9                       141   
AGENT00005                        4                        54   
...                             ...                       ...   
AGENT01196                        7                       102   
AGENT01197                        6                        69   
AGENT01198                        4                        57   
AGENT01199                        8                        78   
AGENT01200                        9                       120   

            POSTAL_CODE_AGENT  EMP_ROUTING_NUMBER  POSTAL_CODE  COLOCATION  \
AGENT_ID                                                                     
AGENT00001              56770           345849580     405556.0           0   
AGENT00002              97212           322091289          0.0           0   
AGENT00003             799711           899185067     353933.0           0   
AGENT00004             388848           810765252     384736.0           0   
AGENT00005             180550           574756585     372593.0           0   
...                       ...                 ...          ...         ...   
AGENT01196             200080           321379380     246818.0           0   
AGENT01197             216642           424780122     222398.0           0   
AGENT01198             149807           874957384     153207.0           0   
AGENT01199             640032           155392472     220662.0           0   
AGENT01200             231407           134496340     236926.0           0   

            CLAIM_DEVIATION  ELIGIBLE_FOR_DISCOUNT  
AGENT_ID                                            
AGENT00001               -2                      1  
AGENT00002                0                      0  
AGENT00003                0                      0  
AGENT00004               

In [60]:
# Fetch low performing agent data
low_performing_agents = Agent_grouped_data[Agent_grouped_data['CLAIM_AMOUNT']<=bottom_5_percentile][['CLAIM_AMOUNT']]
low_performing_agents

CLAIM_AMOUNT
AGENT_ID                
AGENT00006         22400
AGENT00014          7400
AGENT00016         27000
AGENT00085         25700
AGENT00088         27700
AGENT00091         30000
AGENT00098         28200
AGENT00134         22000
AGENT00149         28900
AGENT00156         13200
AGENT00172         22200
AGENT00185         16100
AGENT00186         27700
AGENT00188         28400
AGENT00197         18500
AGENT00252         20000
AGENT00263         11500
AGENT00275         16200
AGENT00279         28000
AGENT00300         12600
AGENT00310         28800
AGENT00316         22200
AGENT00351         21700
AGENT00425         17800
AGENT00442         30300
AGENT00536         20000
AGENT00562         21100
AGENT00571         24200
AGENT00585         16600
AGENT00590         22200
AGENT00604          5500
AGENT00621          5300
AGENT00628          6100
AGENT00672         21300
AGENT00689         13700
AGENT00706         10800
AGENT00721          9000
AGENT00732         11300
AGENT00773         15000
AGENT00780         11300
AGENT00790          6000
AGENT00804         20000
AGENT00805         22600
AGENT00830         23500
AGENT00841         17800
AGENT00856         24200
AGENT00894         27600
AGENT00910         15500
AGENT00969         28300
AGENT00985         30000
AGENT01021         24300
AGENT01037          5600
AGENT01052         28000
AGENT01077          4200
AGENT01107         24700
AGENT01128         28000
AGENT01131         27500
AGENT01151         28000
AGENT01154           800
AGENT01179         27300

# Task #12
* Business wants to find all Suspicious Employees (Agents).
* IF TOTAL CLAIM AMOUNT which meet below criteria is >= 15000 THEN AGENT is classified
as Suspicious ELSE Not
* CLAIM_STATUS = Approved AND CUSTOMER_RISK_SEGMENTATION = High
* AND INCIDENT_SEVERITY = “Major Loss”
* If Suspicious, then 1 ELSE 0. Find mean of this column.

In [62]:
# Create condition in which various criterias are applied for features
condition = (view['CLAIM_STATUS']=='A')&(view['RISK_SEGMENTATION']=='H')&(view['INCIDENT_SEVERITY']=='Major Loss')
condition

0       False
1       False
2       False
3       False
4       False
        ...  
9995    False
9996    False
9997    False
9998    False
9999    False
Length: 10000, dtype: bool

In [63]:
# Transaction that fulfill conditions
condition_fulfilled = view[condition]
condition_fulfilled

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
8       2020-06-01    TXN00000009   A00004671   PLC00006886    2019-06-27   
75      2020-06-03    TXN00000076   A00009078   PLC00001505    2014-08-15   
76      2020-06-03    TXN00000077   A00000804   PLC00006719    2011-07-09   
92      2020-06-04    TXN00000093   A00007477   PLC00008821    2015-06-05   
99      2020-06-04    TXN00000100   A00007283   PLC00001227    2014-09-25   
...            ...            ...         ...           ...           ...   
9826    2021-06-24    TXN00009827   A00006637   PLC00000398    2017-06-26   
9838    2021-06-24    TXN00009839   A00003275   PLC00009400    2013-10-31   
9855    2021-06-25    TXN00009856   A00001546   PLC00000061    2017-01-27   
9948    2021-06-29    TXN00009949   A00003202   PLC00000643    2014-09-02   
9958    2021-06-29    TXN00009959   A00008989   PLC00006292    2018-01-12   

         LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  \
8     2020-05-21  2020-05-26          Motor           82.14          5000   
75    2020-05-17  2020-05-22           Life           67.88         59000   
76    2020-05-14  2020-05-19           Life           68.72         60000   
92    2020-05-23  2020-05-28         Health          116.27          7000   
99    2020-05-30  2020-05-31         Mobile           11.34           200   
...          ...         ...            ...             ...           ...   
9826  2021-06-09  2021-06-14       Property          125.63         13000   
9838  2021-06-09  2021-06-14         Health          199.52          3000   
9855  2021-06-21  2021-06-22          Motor           87.78          4000   
9948  2021-06-24  2021-06-25       Property          113.24         35000   
9958  2021-06-09  2021-06-14         Health          112.88          2000   

      ...     EMP_ACCT_NUMBER                VENDOR_NAME  \
8     ...  VHVZ90875461509264                 Hanson Inc   
75    ...  FMCB37381150648042  Gordon, Young and Webster   
76    ...  BMAQ41191552040977              Walker-Ingram   
92    ...  IGXL71581076733788                Sanchez LLC   
99    ...  CEIG07819775081568                        NaN   
...   ...                 ...                        ...   
9826  ...  OEIB44255461327063                        NaN   
9838  ...  BAJN34272673771226                 Dennis LLC   
9855  ...  MTDS80945137889409                        NaN   
9948  ...  GZAY41352995514708   Reed, Bartlett and Miles   
9958  ...  MWGP08083997541638                Mcclure Ltd   

              ADDRESS_LINE1 ADDRESS_LINE2          CITY  STATE POSTAL_CODE  \
8     910 Arlington Terrace           NaN  Fayetteville     AR     72701.0   
75           82 Queen Court           NaN    Manchester     CT      6040.0   
76          4770 Balboa Way           NaN       Fremont     CA     94536.0   
92        154 Lyness Street           NaN    Manchester     CT      6040.0   
99                      NaN           NaN           NaN    NaN         NaN   
...                     ...           ...           ...    ...         ...   
9826                    NaN           NaN           NaN    NaN         NaN   
9838      511 Norton Avenue           NaN     Nashville     TN     37207.0   
9855                    NaN           NaN           NaN    NaN         NaN   
9948   37 East Broad Street           NaN      Savannah     GA     31401.0   
9958    243 Kentucky Avenue           NaN      Pasadena     MD     21122.0   

     COLOCATION  CLAIM_DEVIATION  ELIGIBLE_FOR_DISCOUNT  
8             0                0                      0  
75            0                0                      0  
76            0                0                      0  
92            0                0                      0  
99            0                0                      0  
...         ...              ...                    ...  
9826          0               -1                      0  
9838          0               -1 

In [64]:
# Joining condition fulfilled with AGENT_ID feature
grouped_suspicion_data = condition_fulfilled.groupby('AGENT_ID').sum()
grouped_suspicion_data

PREMIUM_AMOUNT  CLAIM_AMOUNT  POSTAL_CODE_CUSTOMER  AGE  TENURE  \
AGENT_ID                                                                      
AGENT00001          135.12         39000                 72701   37     108   
AGENT00003           80.30         44000                 80002   50      77   
AGENT00007           66.92          3000                  1027   30      41   
AGENT00011          211.88         48200                289139  146     197   
AGENT00013           64.53          1000                 36106   62      97   
...                    ...           ...                   ...  ...     ...   
AGENT01184          199.36         38100                 61435  110      79   
AGENT01186           85.31         10000                 73159   26      34   
AGENT01188           79.42         53000                 94551   45      38   
AGENT01197          179.50         62000                105196   80     208   
AGENT01200          223.01         14000                132905   68     146   

            NO_OF_FAMILY_MEMBERS  ROUTING_NUMBER  ANY_INJURY  \
AGENT_ID                                                       
AGENT00001                     7        79664480           1   
AGENT00003                     2        81620470           1   
AGENT00007                     4        37315551           1   
AGENT00011                    14       263729738           1   
AGENT00013                     4        99394222           0   
...                          ...             ...         ...   
AGENT01184                     7       149113307           1   
AGENT01186                     7        13941538           0   
AGENT01188                     5        99711287           0   
AGENT01197                    13        83880466           2   
AGENT01200                    12       192648642           1   

            POLICE_REPORT_AVAILABLE  INCIDENT_HOUR_OF_THE_DAY  \
AGENT_ID                                                        
AGENT00001                        1                         6   
AGENT00003                        0                         7   
AGENT00007                        1                        11   
AGENT00011                        1                        22   
AGENT00013                        0                        18   
...                             ...                       ...   
AGENT01184                        2                        44   
AGENT01186                        1                        13   
AGENT01188                        1                        23   
AGENT01197                        2                        23   
AGENT01200                        1                        11   

            POSTAL_CODE_AGENT  EMP_ROUTING_NUMBER  POSTAL_CODE  COLOCATION  \
AGENT_ID                                                                     
AGENT00001               5677            34584958      40207.0           0   
AGENT00003              72701            81744097      32413.0           0   
AGENT00007              85302            53819552      37207.0           0   
AGENT00011              60057            85724709          0.0           0   
AGENT00013              94608            22235031       6042.0           0   
...                       ...                 ...          ...         ...   
AGENT01184             111618           161200914          0.0           0   
AGENT01186              37210            57617352          0.0           0   
AGENT01188              80007            45452985      80005.0           0   
AGENT01197              72214           141593374     104652.0           0   
AGENT01200              42074            24453880      42157.0           0   

            CLAIM_DEVIATION  ELIGIBLE_FOR_DISCOUNT  
AGENT_ID                                            
AGENT00001                0                      0  
AGENT00003                0                      0  
AGENT00007               -1                      0  
AGENT00011               

In [65]:
# Storing agents_id that are suspicious (met all the condition)
suspicious_agents = list(grouped_suspicion_data[grouped_suspicion_data['CLAIM_AMOUNT']>=15000].index)
suspicious_agents

['AGENT00001',
 'AGENT00003',
 'AGENT00011',
 'AGENT00021',
 'AGENT00043',
 'AGENT00052',
 'AGENT00076',
 'AGENT00080',
 'AGENT00097',
 'AGENT00112',
 'AGENT00136',
 'AGENT00143',
 'AGENT00144',
 'AGENT00145',
 'AGENT00152',
 'AGENT00155',
 'AGENT00158',
 'AGENT00168',
 'AGENT00176',
 'AGENT00196',
 'AGENT00199',
 'AGENT00207',
 'AGENT00210',
 'AGENT00214',
 'AGENT00222',
 'AGENT00224',
 'AGENT00241',
 'AGENT00250',
 'AGENT00251',
 'AGENT00260',
 'AGENT00284',
 'AGENT00299',
 'AGENT00302',
 'AGENT00307',
 'AGENT00309',
 'AGENT00313',
 'AGENT00314',
 'AGENT00315',
 'AGENT00324',
 'AGENT00348',
 'AGENT00381',
 'AGENT00388',
 'AGENT00390',
 'AGENT00396',
 'AGENT00398',
 'AGENT00399',
 'AGENT00402',
 'AGENT00423',
 'AGENT00432',
 'AGENT00457',
 'AGENT00458',
 'AGENT00462',
 'AGENT00470',
 'AGENT00474',
 'AGENT00482',
 'AGENT00501',
 'AGENT00513',
 'AGENT00521',
 'AGENT00524',
 'AGENT00525',
 'AGENT00528',
 'AGENT00529',
 'AGENT00533',
 'AGENT00535',
 'AGENT00544',
 'AGENT00565',
 'AGENT005

In [66]:
# Read employee full dataframe as done earlier before moving to task
employee

AGENT_ID         AGENT_NAME DATE_OF_JOINING  \
0     AGENT00001          Ray Johns      1993-06-05   
1     AGENT00002      Angelo Borjon      2005-12-27   
2     AGENT00003     Candy Spellman      2003-09-02   
3     AGENT00004         Mary Smith      2004-09-23   
4     AGENT00005       Mildred Diaz      2011-06-21   
...          ...                ...             ...   
1195  AGENT01196     Myrtle Mccatty      2005-09-14   
1196  AGENT01197        John Dorsey      1993-12-04   
1197  AGENT01198        Ruth Bigbee      1993-09-05   
1198  AGENT01199     Tommie Camacho      2004-10-25   
1199  AGENT01200  Virginia Stachura      1999-05-25   

                         ADDRESS_LINE1 ADDRESS_LINE2              CITY STATE  \
0                     1402 Maggies Way           NaN  Waterbury Center    VT   
1                       414 Tanya Pass           NaN       Panama City    FL   
2                  606 National Street          #306      Fayetteville    AR   
3                235 Hugh Thomas Drive           NaN       Panama City    FL   
4                3426 Broadview Street           NaN        Montgomery    AL   
...                                ...           ...               ...   ...   
1195  3553 Brandywine Street Northwest           NaN        Washington    DC   
1196              2315 Chestnut Street           NaN        Montgomery    AL   
1197            1983 Reidsville Street           NaN         Annapolis    MD   
1198                  5821 Taft Street           NaN            Arvada    CO   
1199                   3579 2nd Avenue           NaN         Edgewater    MD   

      POSTAL_CODE  EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  
0            5677            34584958  HKUN51252328472585  
1           32404           107363763  OPIS19290040088204  
2           72701            81744097  YSCJ67489688482590  
3           32404            67563771  ZANG21285355574581  
4           36110           114951317  DZFS82244494451134  
...           ...                 ...                 ...  
1195        20008            32137938  EWBR19907574832501  
1196        36107            70796687  KGEJ74846009867837  
1197        21401           124993912  RASV88064214164820  
1198        80004            19424059  UOFO11656245339211  
1199        21037            12226940  ALYY24945545778511  

[1200 rows x 10 columns]

In [67]:
# Make values of SUSPICIOUS feature as 0
employee['SUSPICIOUS'] = 0
employee['SUSPICIOUS']

0       0
1       0
2       0
3       0
4       0
       ..
1195    0
1196    0
1197    0
1198    0
1199    0
Name: SUSPICIOUS, Length: 1200, dtype: int64

In [68]:
# Fetch suspicious employee based on their AGENT_ID 
suspicious_employee_slice = employee['AGENT_ID'].isin(suspicious_agents)
suspicious_employee_slice

0        True
1       False
2        True
3       False
4       False
        ...  
1195    False
1196     True
1197    False
1198    False
1199    False
Name: AGENT_ID, Length: 1200, dtype: bool

In [69]:
# Changing suspicious feature value to 1 of suspicious employees
employee.loc[suspicious_employee_slice,'SUSPICIOUS']=1
employee['SUSPICIOUS']

0       1
1       0
2       1
3       0
4       0
       ..
1195    0
1196    1
1197    0
1198    0
1199    0
Name: SUSPICIOUS, Length: 1200, dtype: int64

In [70]:
# Fetch the mean value for the feature 'SUSPICIOUS'
employee['SUSPICIOUS'].mean()

0.12416666666666666

Conclusion Drawn:-

The mean is 0.1241 so it means that there are 12.41% employees in SUSPICIOUS feature.